module/zfs/dsl_scan.c

```diff
@@ -1726,6 +1726,13 @@ dsl_scan_check_resume(dsl_scan_t *scn, const dnode_phys_t *dnp,
 	 */
 	if (!ZB_IS_ZERO(&scn->scn_phys.scn_bookmark) &&
 	    (int64_t)zb->zb_object >= 0) {
+		/*
+		 * If we stopped at the root level we haven't scanned
+		 * anything yet.
+		 */
+		if (scn->scn_phys.scn_bookmark.zb_level == ZB_ROOT_LEVEL)
+			return (B_FALSE);
```

Comment:
If you have a pool that's in the middle of a scrub that has suspended at the objset (ZB_ROOT_LEVEL), and you open that pool with older software, this code won't be present.  What happens then?  Seems like the code below might do the wrong thing.

In that case, we need to either add an on-disk feature flag for this, or do something more tricky (e.g. never write ZB_ROOT_LEVEL to disk, only allowing it to impact in-memory behavior).